kedro-org/kedro-viz
# Add UnavailableDataset as a default dataset for lite mode

## Description

Related to #2058 

## Development notes

* Created a custom dataset `UnavailableDataset`
* Updated the default dataset from `MemoryDataset` to `UnavailableDataset` in `--lite` mode.
* Updated filepath to have an empty string as `N/A` instead of `-` as suggested by @stephkaiser 
* Updated tests

## QA notes

* When you run `kedro viz --lite` with `kedro-datasets` missing in the conda env, the metadata panel for the dataset should show the dataset type as `UnavailableDataset` if the dataset is mentioned in catalog.py (i.e., not a MemoryDataset)

<img width="389" alt="Screenshot 2024-09-05 at 2 43 30 PM" src="https://github.com/user-attachments/assets/ff0cb35a-04c9-46c6-b872-a06b7d67cf05">


* All tests should pass


## Checklist
